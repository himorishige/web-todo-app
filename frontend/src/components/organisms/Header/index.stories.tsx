/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Header from '.';

export default {
  title: 'WebToDoApp/Organisms/Header',
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Header';
