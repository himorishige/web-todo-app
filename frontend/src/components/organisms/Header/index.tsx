/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { BsTable } from 'react-icons/bs';
import { Link } from 'react-router-dom';
import { StarIcon } from 'src/components/atoms';
import { useAppDispatch, useAppSelector } from 'src/app/hooks';
import { selectStarStatus, toggleFilter } from 'src/features/tasks/tasksSlice';

type Props = {};

const Header: React.VFC<Props> = () => {
  const dispatch = useAppDispatch();
  const starState = useAppSelector(selectStarStatus);

  const toggleHandler = () => {
    dispatch(toggleFilter());
  };

  return (
    <div css={titleStyle}>
      <div css={innerStyle}>
        <div>
          <Link to="/">
            <BsTable css={iconStyle} />
          </Link>
        </div>
        <h1 css={headingStyle}>ToDo App</h1>
        <div css={starToggleStyle} onClick={toggleHandler}>
          <StarIcon status={+starState} />
        </div>
      </div>
    </div>
  );
};

export default Header;

const titleStyle = css`
  background: linear-gradient(to left, rgba(216, 179, 132, 1), rgba(216, 179, 132, 0.7));
`;

const innerStyle = css`
  max-width: 768px;
  margin: 0 auto;
  padding: 0.75rem 1.25rem;
  display: flex;
  align-items: center;
  justify-content: space-between;

  a {
    color: var(--black-color);
    transition: opacity 0.3s;
    &:hover {
      opacity: 0.8;
    }
  }
`;

const headingStyle = css`
  font-family: 'Poppins', sans-serif;
  margin: 0;
`;

const iconStyle = css`
  width: 2rem;
  height: 2rem;
  margin-right: 0.5rem;
`;

const starToggleStyle = css`
  svg {
    width: 2rem;
    height: 2rem;
  }
`;
