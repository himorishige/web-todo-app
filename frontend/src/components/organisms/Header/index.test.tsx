import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from 'src/app/store';
import Header from '.';

describe('Organisms/Header', () => {
  test('正しくレンダリングされている', () => {
    const element = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Header />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });
});
