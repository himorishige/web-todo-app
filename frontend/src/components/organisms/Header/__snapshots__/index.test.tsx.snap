// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Header 正しくレンダリングされている 1`] = `
<div
  className="css-1hrd1c8"
>
  <div
    className="css-1hh8z22"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div>
        <svg
          className="css-1443oa1"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M14 1H2a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V2a1 1 0 00-1-1zM2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M15 4H1V3h14v1z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M5 15.5v-14h1v14H5zm5 0v-14h1v14h-1z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M15 8H1V7h14v1zm0 4H1v-1h14v1z"
            fillRule="evenodd"
          />
          <path
            d="M0 2a2 2 0 012-2h12a2 2 0 012 2v2H0V2z"
          />
        </svg>
      </div>
    </a>
    <h1
      className="css-1llq2fx"
    >
      ToDo App
    </h1>
    <div
      className="css-knz8lr"
      data-testid="toggle-button"
      onClick={[Function]}
    >
      <div
        className="css-kknodv"
      >
        <svg
          className="css-ba990d"
          data-testid="star"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.523-3.356c.329-.314.158-.888-.283-.95l-4.898-.696L8.465.792a.513.513 0 00-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767l-3.686 1.894.694-3.957a.565.565 0 00-.163-.505L1.71 6.745l4.052-.576a.525.525 0 00.393-.288l1.847-3.658 1.846 3.658a.525.525 0 00.393.288l4.052.575-2.906 2.77a.564.564 0 00-.163.506l.694 3.957-3.686-1.894a.503.503 0 00-.461 0z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
